documentation/projects/proposals/document_all_media_properties/20230426-implementation_plan_documenting_media_properties.md

```diff
@@ -0,0 +1,137 @@
+
+- [ ] @AetherUnbound
+- [ ] @stacimc
+
+## Project links
+
+<!-- Enumerate any references to other documents/pages, including milestones and other plans -->
+
+- [Project Thread](https://github.com/WordPress/openverse/issues/412)
+- [Project Proposal](https://docs.openverse.org/projects/proposals/document_all_media_properties/20230307-project_proposal.html#description)
+
+## Overview
+
+<!-- A brief one or two sentence overview of the implementation being described. -->
+
+A proposal to create a documentation page with description of the media
+properties collected by the Catalog.
+
+## Background
+
+The catalog collects and transforms data from many different sources. The kind
+of data collected, transformations, and the expected data shape have changed
+over time. This plan will outline the creation of the documentation page about
+the media properties in the catalog. This will help answer some of the questions
+like:
+
+### User stories
+
+- As a maintainer/contributor, I need to know the list of available properties
+  for each media type in the catalog database. ... so that?
+- As a contributor, I need to know what is the expected data type for the
+  specific property. Is it required? ... so that?
+- As a contributor, I need to know how the data property is transformed and
+  validated.
+- As a contributor writing a provider API script, which data properties do I
+  need to collect?
+- As a contributor writing a provider API script, how do I select data for the
+  specific media property Openverse uses?
+- As a contributor writing a provider script, which format should I convert a
+  media property?
+- As a maintainer, I've found a discrepancy between the type of data property
+  the API returns and what is described in the table (e.g., is `null` for a
+  non-nullable property). Is there an issue to fix this data in the catalog
+  database, or do I need to create a new one?
+- As an API user, I need to know what exactly this media property is and what
+  data points from the providers are used for it. E.g., does the "creator" refer
+  to the person who created the object or took the picture?
+- As a provider of openly-licensed media, I want to know what media properties
+  Openverse collects so that I can change my API responses.
+
+The information on data types, transformations, and validations will be
+presented as a table and will be extracted from the code. Other information
+about the data selection properties and data discrepancies will be written in a
+separate markdown file as it is easier to read, write, and lint.
+
+The catalog has a DAG documentation generator that extracts the docstrings from
+the DAGs and creates a page with the documentation. We can use the same approach
+to create a documentation page for the media properties. The DAG doc generator
+runs the code inside docker to extract DAG information. In this project, we can
```

Comment:
Small potential wording change here for clarity:
```suggestion
to create a documentation page for the media properties. The DAG doc generator
runs the code inside docker to extract DAG information because it requires the project dependencies (i.e. `airflow`) in order to parse the DAG. In this project, we can
```